.github/workflows/build-package.yml

```diff
@@ -38,18 +38,30 @@ jobs:
         password: ${{ secrets.GITHUB_TOKEN }}
 
     steps:
+    - name: Get correct ref
+      run: echo "GH_REF=${{ github.event_name == 'pull_request' }} && ${{ github.head_ref }} || ${{ github.ref_name }}" >> $GITHUB_ENV
+    
+    - name: Activate environment
+      run: spack env activate ${{ inputs.package-name }}
+    
     - name: Build package via spack
+      run: spack -d install --add --only package --fail-fast --no-checksum ${{ inputs.package-name }}@=$GH_REF%${{ inputs.compiler-name }}@${{ inputs.compiler-version }}
+    
+    - name: Generate lockfiles on failure
```

Comment:
We don't want to generate lockfile regardless of installation success or failure? I don't exactly know what I want to do with them though. :)

Reply:
Might be worth doing them on both conditions, actually. Will update the repo.

Reply:
Added on both conditions